Add a service to trigger functionality (#1611)
gazebosim/gz-sim: tutorials/triggered_publisher.md

\page triggeredpublisher Triggered Publisher

The `TriggeredPublisher` system publishes a user specified message on an output
topic in response to an input message that matches user specified criteria. It

> can also call a user specified service as an output in response to an input

Reviewer: Nit: A little confusing on first read
```suggestion
can also call a user specified service in response to an input
```

\page triggeredpublisher Triggered Publisher

The `TriggeredPublisher` system publishes a user specified message on an output
topic or service in response to an input message that matches user specified criteria.

> The system works by checking the input against a set of Matchers. Matchers

Reviewer: Same wording change as above.

cause a box to fall from its initial position by detaching a detachable joint
in response to the motion of a vehicle. The tutorial also covers how Triggered
Publisher systems can be chained together by showing how the falling of the box

> can trigger another box to fall. Last, it'll cover how service call can be triggered
> to reset the robot pose. The finished world SDFormat file for this

Reviewer: Consistent verb tense
```suggestion
can trigger another box to fall. Last, it covers how a service call can be triggered
to reset the robot pose. The finished world SDFormat file for this
```

```
ign topic -t "/start" -m ignition.msgs.Empty -p " "
```

Once both boxes have fallen, we can publish a message to invoke a service call
to reset the robot position as well as set the speed to 0. As shown below, the

> `<output>` sets the linear x speed to 0 and `<service>` tag contains metadata

Reviewer: Nit
```suggestion
`<output>` sets the linear x speed to 0, and the `<service>` tag contains metadata
```

to invoke a service call to `/world/triggered_publisher/set_pose`. The `reqMsg`
is expressed in the human-readable form of Google Protobuf meesages. Multiple

> services can be used and it can also be used with the `<output>` tag.

Reviewer: What "it" refers to here is unclear.

```xml
<plugin filename="ignition-gazebo-triggered-publisher-system"
  name="ignition::gazebo::systems::TriggeredPublisher">
  <input type="ignition.msgs.Empty" topic="/reset_robot"/>
  <output type="ignition.msgs.Twist" topic="/cmd_vel">
      linear: {x: 0}
  </output>
  <service name="/world/triggered_publisher/set_pose"
    reqType="ignition.msgs.Pose" repType="ignition.msgs.Boolean"

>       timeout="3000" reqMsg="name: 'blue_vehicle', id:8, position: {x:-3, z:1}">

Reviewer: ```suggestion
      timeout="3000" reqMsg="name: 'blue_vehicle', id: 8, position: {x: -3, z: 1}">
```
